### PR TITLE
Refactored evaluation

### PR DESCRIPTION
**Pull request by laurikskl**
Refactored evaluation.

Sorted out exports and imports.

Also broke up the evaluate methods to several smaller functions to reduce CC and improve readability.

### CODE REVIEW

File: src/evaluation/Evaluation.jl [hunk 1]
```diff
@@ -5,10 +5,17 @@ using ..CounterfactualExplanations.Objectives
...
+using ProgressMeter
+using LinearAlgebra
+using SliceMap
 
 export Benchmark, benchmark, evaluate, default_measures
 export validity, distance, redundancy
 
+include("benchmark.jl")
+include("evaluate.jl")
 include("measures.jl")
```
Comment (kmariuszk) [inline src/evaluation/Evaluation.jl:20]: In your other PR (#229) you used the following order: `using`, `include`, `export` but here `export` and `include` are the other way around. Could you please change it so we'd end up using the same convention in all files?

**Review by kmariuszk — CHANGES_REQUESTED** [review]

**Review by kmariuszk — APPROVED** [review]

File: src/evaluation/benchmark.jl [hunk 2]
```diff
@@ -1,8 +1,3 @@
-using DataFrames
-using ..Models: train
-using ProgressMeter
-using Statistics
-
 "A container for benchmarks of counterfactual explanations. Instead of subtyping `DataFrame`, it contains a `DataFrame` of evaluation measures (see [this discussion](https://discourse.julialang.org/t/creating-an-abstractdataframe-subtype/36451/6?u=pat-alt) for why we don't subtype `DataFrame` directly)."
 struct Benchmark
     evaluation::DataFrame
```
Comment (VincentPikand) [inline src/evaluation/benchmark.jl:3]: `DataFrames.DataFrame`

File: src/evaluation/evaluate.jl [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+function evaluate_dataframe(
+    ce::CounterfactualExplanation,
+    measure::Vector{Function},
+    agg::Function,
+    report_each::Bool,
+    pivot_longer::Bool,
+    store_ce::Bool,
+)
+    computed_measures = [compute_measure(ce, fun, agg) for fun in measure]
+    evaluation = DataFrame(
```
Comment (VincentPikand) [inline src/evaluation/evaluate.jl:46]: no prefix on DataFrame

File: src/evaluation/evaluate.jl [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        Dict(
+            m => report_each ? val[1] : val for
+            (m, val) in zip(Symbol.(measure), computed_measures)
+        ),
+    )
+    evaluation.num_counterfactual = 1:nrow(evaluation)
+    if pivot_longer
+        evaluation = stack(evaluation, Not(:num_counterfactual))
```
Comment (VincentPikand) [inline src/evaluation/evaluate.jl:54]: no prefix on `Not`

File: src/evaluation/evaluate.jl [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+"""
+    generate_meta_data(i::Int, ce::CounterfactualExplanation, evaluation::DataFrame, report_meta::Bool, meta_data::Union{Nothing,Vector{Dict}})
+Generates meta data for a counterfactual explanation. If `report_meta=true`, the meta data is extracted from the counterfactual explanation. If `meta_data` is supplied, it is used instead.
+"""
+
+function generate_meta_data(
+    i::Int,
+    ce::CounterfactualExplanation,
+    evaluation::DataFrame,
```
Comment (VincentPikand) [inline src/evaluation/evaluate.jl:71]: as mentioned above

File: src/evaluation/measures.jl [hunk 6]
```diff
@@ -1,13 +1,9 @@
...
-
 """
     validity(ce::CounterfactualExplanation; γ=0.5)
 
 Checks of the counterfactual search has been successful with respect to the probability threshold `γ`. In case multiple counterfactuals were generated, the function returns the proportion of successful counterfactuals.
 """
-function validity(ce::CounterfactualExplanation; agg=mean, γ=0.5)
+function validity(ce::CounterfactualExplanation; agg=Statistics.mean, γ=0.5)
     val = agg(CounterfactualExplanations.target_probs(ce) .>= γ)
     val = val isa AbstractMatrix ? vec(val) : val
```
Comment (VincentPikand) [inline src/evaluation/measures.jl:8]: `AbstractMatrix` not from `Base`

**Review by VincentPikand — CHANGES_REQUESTED** [review]
please go over everything again and add all namespace prefixes, I didn't comment on every single one I found.

**Comment by laurikskl** [comment]
> please go over everything again and add all namespace prefixes, I didn't comment on every single one I found.

done

**Review by kmariuszk — APPROVED** [review]

**Review by VincentPikand — APPROVED** [review]

**Pull request merged**